Add Plone 5.1 Support

busykoala opened the pull request:
This pull changes:
- Plone 5.1 way of register resources as bundles in registry.xml
- Adds 5.1 test cfg
- Disregards DOM attribute order when checking for the attributes and values
- Manually add csrf token in plone 5.1 test to enable json view saving

busykoala commented on ftw/iframeblock/tests/test_builder.py:
```diff
@@ -12,11 +13,22 @@ def setUp(self):
+                             'must be the only child element of the content '
+                             'page having the title given.')
+        else:
+            self.assertTrue(len(browser.css('.sl-block')), 'Expect one block')
```
Here in Plone 4.3 we find a `sl-block` in the DOM. Now in Plone 5.1 for some reason the view is never called (and not rendered). The test I implemented above having the `IS_PLONE_5` condition works and therefore shows, that the node obj exists.
Do you have any Idea why the view is never called?

maethu replied:
It could have a render issue? If a block fails to render it dies silently. Maybe check how it does look on a booted system.

busykoala replied:
On the booted system it is working and also has the same css selector checked for in the test.

![image](https://user-images.githubusercontent.com/29920936/55547201-18753980-56d1-11e9-8b22-0a00d9a50ce2.png)

maethu replied:
This is weird...

busykoala replied:
It seems that this was an issue of the properties which were not registered. The test as it was before works now again and the block is rendered.

maethu commented on ftw/iframeblock/profiles/default_plone5/propertiestool.xml:
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0"?>
```
This does not work in plone5. Translate/Move to the equivalent `registry.xml` records.

busykoala replied:
👍 I added the resources to registry.xml

busykoala commented:
@maethu could you please look at this again? I added the properties to registry now for plone 5 and this also fixed the rendering issue we had earlier. Therefore I undid the change in this test going for the css selector.

maethu commented on ftw/iframeblock/tests/test_content_type.py:
```diff
@@ -26,10 +26,15 @@ def test_block_can_be_added_with_factories_menu(self, browser):
+        tree = html.fromstring(browser.contents)
+        iframe = tree.xpath('//iframe')[0]
+
+        self.assertEqual(iframe.get('width'), '100%')
```
Everything looks good now, but why do you don't use ftw.testbrowser features for the assertions?
like `browser.css('iframe').first.attrib['css], 'iframeblock loading')` or `self.assertTrue(browser.css('iframe.iframeblock.loading')` ?

maethu replied:
ftw.testbrowser already does the lxml part for you

busykoala replied:
Ich kannte `.attrib` nicht und kann besser antizipieren was das Resultat sein wird, wenn ich xpath brauche. Ich habe das der Leserlichkeit halber korrigiert.
Zudem habe ich `assertListEqual` eingeführt, nachdem ich die verschiedenen Parameter in eine sorted List eingefügt habe. So ist im Diff gleich ersichtlich, wo der Fehler liegt.

maethu commented:
💛
